frontend/src/Components/HomePage/AddTrip/FlightForm.jsx

```diff
@@ -0,0 +1,258 @@
+
+
+    const handleSubmit = async (e) => {
+
+        e.preventDefault();
+        setFlightStatus(null);
+        setError(null);
+        setIsLoading(true);
+        setFlightAdded(false);  
+        
+        const timeoutDuration = 60000;
+        const controller = new AbortController();
+        const timeoutId = setTimeout(() => controller.abort(), timeoutDuration);
+    
+        try {
+            const response = await axios.get(`${API_BASE_URL}/flight-status`, {
+            params: {
+                carrierCode: carrierCode,
+                flightNumber: flightNumber,
+                scheduledDepartureDate: departureDate
+            }
+            });
+            clearTimeout(timeoutId);
+            
+            if (response.data && response.data.data && response.data.data.length > 0) {
+                setFlightStatus(response.data);
+            } else {
+                setError("No flights found. Please check your input and try again.");
+            }
+        
+        } catch (err) {
+            clearTimeout(timeoutId);
+            if (err.name === 'AbortError') {
+                setError("Request timed out. Please try again.");
+            } else if (err.response) {
+                setError(`Error: ${err.response.data.message || 'Failed to fetch flight status'}`);
+            } else if (err.request) {
+                setError("No response received from server. Please check your internet connection.");
+            } else {
+                setError("An unexpected error occurred. Please try again later.");
+            }
+        
+        } finally {
+            setIsLoading(false);
+        }
+        };
+    
+    
+    /**
+     * Adds static trip information to the user's account in the database.
+     * It uses the flight information obtained from the previous search.
+     * 
+     * @throws {Error} If adding the trip fails
+     * 
+     * Note: Real-time flight status should be fetched separately when needed,
+     * as this stored data does not update automatically.
+     */
+
+
+    const handleAddFlight = async () => {
+
+        if (!flightStatus || !flightStatus.data || flightStatus.data.length === 0) {
+            setError("No flight data available to add.");
+            return;
+        }      
+            
+        const flight = flightStatus.data[0];
+        const userString = localStorage.getItem('user');
+        
+        
+        if (!userString) {
+            navigate('/login');
+            return;
+        }
+        
+        const user = JSON.parse(userString);
+        
+        if (!user || !user.userId) {
+            setError("User information is invalid. Please log in again.");
+            navigate('/login');
+            return;
+        }
+        
+        const userId = parseInt(user.userId); 
+        
+        if (isNaN(userId)) {
+            setError("Invalid user ID. Please log in again.");
+            navigate('/login');
+            return;
+        }
```

Comment:
So we are doing this to verify if the user is logged in before making the HTTP Request? This should be verified in the backend, so please make sure you have a check in your backend to validate user session every time you access an API that is supposed to be protected.

If the backend returns an authentication error (or the session is invalid), them you redirect to /login page.